presto-main/src/main/java/com/facebook/presto/operator/ArrayPositionLinks.java

```diff
@@ -46,9 +50,25 @@ public int link(int left, int right)
         }
 
         @Override
-        public Factory build()
+        public PositionLinks.Factory build()
         {
-            return filterFunction -> new ArrayPositionLinks(positionLinks);
+            return new PositionLinks.Factory()
+            {
+                @Override
+                public PositionLinks create(Optional<JoinFilterFunction> joinFilterFunction)
+                {
+                    return new ArrayPositionLinks(positionLinks);
+                }
+
+                @Override
+                public long checksum()
+                {
+                    Hasher hasher = Hashing.md5().newHasher();
```

Comment:
Some other function which is less expensive to compute could be equally good for the purpose.
Though I checked and MD5 seems to work fast. So you may ignore it.